lib/ansible/modules/cloud/azure/azure_rm_containerinstance.py

```diff
@@ -0,0 +1,429 @@
+        # command (list of str)
+        # ports (list of ContainerPort)
+        # environment_variables (list of EnvironmentVariable)
+        # resources (ResourceRequirements)
+        # volume mounts (list of VolumeMount)
+    )
+
+    if container.instance_view is not None:
+        # instance_view (ContainerPropertiesInstanceView)
+        results["instance_restart_count"] = container.instance_view.restart_count
+        if container.instance_view.current_state:
+            results["instance_current_state"] = container.instance_view.current_state.state
+            results["instance_current_start_time"] = container.instance_view.current_state.start_time
+            results["instance_current_exit_code"] = container.instance_view.current_state.exit_code
+            results["instance_current_finish_time"] = container.instance_view.current_state.finish_time
+            results["instance_current_detail_status"] = container.instance_view.current_state.detail_status
+        if container.instance_view.previous_state:
+            results["instance_previous_state"] = container.instance_view.previous_state.state
+            results["instance_previous_start_time"] = container.instance_view.previous_state.start_time
+            results["instance_previous_exit_code"] = container.instance_view.previous_state.exit_code
+            results["instance_previous_finish_time"] = container.instance_view.previous_state.finish_time
+            results["instance_previous_detail_status"] = container.instance_view.previous_state.detail_status
+        # events (list of ContainerEvent)
+    return results
+
+
+class AzureRMContainerInstance(AzureRMModuleBase):
+    """Configuration class for an Azure RM container instance resource"""
+
+    def __init__(self):
+        self.module_arg_spec = dict(
+            resource_group=dict(
+                type='str',
+                required=True
+            ),
+            name=dict(
+                type='str',
+                required=True
+            ),
+            os_type=dict(
+                type='str',
+                default='linux',
+                choices=['linux', 'windows']
+            ),
+            state=dict(
+                type='str',
+                default='present',
+                choices=['present', 'absent']
+            ),
+            location=dict(
+                type='str',
+            ),
+            ip_address=dict(
+                type='str',
+                default='none',
+                choices=['public', 'none']
+            ),
+            ports=dict(
+                type='list',
+                default=[]
+            ),
+            registry_login_server=dict(
+                type='str',
+                default=None
+            ),
+            registry_username=dict(
+                type='str',
+                default=None
+            ),
+            registry_password=dict(
+                type='str',
+                default=None
```

Comment:
Should set `no_log=True` for fields like this, this stops Ansible from logging secret information to plaintext sources.